src/carts/src/carts-service/handlers.go

```diff
@@ -66,7 +72,10 @@ func CartUpdate(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	t := RepoUpdateCart(cart)
+	id := path.Base(r.URL.Path)
```

Comment:
Ah that's much nicer, I've literally never programmed in Go before so this is all new to me.

Reply:
> Also, I think this function could be further improved with some validation that returns HTTP 400 if the cartID in the path does not equal cart.ID from the payload.

Regarding this, if we are going to do a PUT using an ID in the path then I don't think the client should include an id in the payload at all. An alternative is to get rid of the path id parameter all together and just accept the payload with an id (it would work similarly to POST does except we are providing an id). From REST APIs I've worked on in the past, the first approach is the one I've used most often.